src/compute/src/render/join/mz_join_core.rs

```diff
@@ -468,6 +477,10 @@ where
             }
         }
 
+        if !temp.is_empty() {
+            flush(temp, &mut session);
```

Comment:
We should give `fuel` back here as well, no?

Reply:
Yup, that's right!